src/common_interface/solve.jl

```diff
@@ -62,6 +62,8 @@ function DiffEqBase.__init(
 
     progress && @logmsg(-1,progress_name,_id=_id = :Sundials,progress=0)
 
+    uBottomEltype = recursive_bottom_eltype(prob.u0)
```

Comment:
even better: we know it has to be Float64

Reply:
Lol let me redo this whole PR